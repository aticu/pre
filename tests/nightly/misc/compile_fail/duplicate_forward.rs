use pre::pre;

fn foo() {}

mod nested {
    use pre::pre;

    #[pre("nested foo")]
    pub(super) fn foo() {}
}

mod other_nested {
    use pre::pre;

    #[pre("nested foo")]
    pub(super) fn foo() {}
}

#[pre]
fn main() {
    #[forward(nested)]
    #[forward(other_nested)]
    #[assure("nested foo", reason = "corresponding forward present")]
    foo();
}
