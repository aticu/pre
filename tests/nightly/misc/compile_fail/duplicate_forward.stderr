error: duplicate `forward` attribute
  --> $DIR/duplicate_forward.rs:22:5
   |
22 |     #[forward(other_nested)]
   |     ^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: there can be just one location, try removing the wrong one
  --> $DIR/duplicate_forward.rs:21:5
   |
21 |     #[forward(nested)]
   |     ^^^^^^^^^^^^^^^^^^
